tests/lib/pkgdb.sh

```diff
@@ -850,3 +850,86 @@ distro_upgrade() {
             ;;
     esac
 }
+
+distro_list_packages() {
+    where="$1"
+
+    case "$SPREAD_SYSTEM" in
+        fedora-*|centos-*|amazon-linux-*|opensuse-*)
+            rpm -qa | sort > "$where"
+            ;;
+        ubuntu-*|debian-*)
+            dpkg --get-selections > "$where"
+            ;;
+        arch-*)
+            pacman -Q | cut -f1 -d' ' | sort > "$where"
+            ;;
+        *)
+            ;;
+    esac
+}
+
+distro_restore_packages() {
+    before="$1"
+    after="$before.after"
+    installed="$before.installed"
+    removed="$before.removed"
+
+    # find out what's installed now on systems where we the old list is
+    # not enough
+    case "$SPREAD_SYSTEM" in
+        fedora-*|centos-*|amazon-linux-*|opensuse-*)
+            rpm -qa | sort > "$after"
+            ;;
+        arch-*)
+            # the output format of pacman -Q is:
+            # zsh 5.7.1-1\n
+            pacman -Q | cut -f1 -d' ' | sort > "$after"
+            ;;
+        ubuntu-*|debian-*)
+            dpkg --get-selections > "$after"
+            ;;
+    esac
```

Comment:
Can you re-use distro_list_packages here?

Reply:
Yeah, this should work.